test/browser-reporters/html.spec.js

```diff
@@ -0,0 +1,1918 @@
+'use strict';
+
+var HTML = require('../../lib/reporters/html');
+
+var Suite = require('../../lib/suite');
+var Context = require('../../lib/context');
+var Test = require('../../lib/test');
+
+describe('HTML reporter', function () {
+  this.slow(250);
+
+  var realOutput;
+  var realOutputStats;
+  var realOutputReport;
+
+  function saveRealOutput () {
+    if (!realOutput) {
+      realOutput = document.getElementById('mocha');
+      realOutputStats = document.getElementById('mocha-stats');
+      realOutputReport = document.getElementById('mocha-report');
+    }
+  }
+
+  function removeRealOutput () {
+    saveRealOutput();
+    if (realOutput) {
+      realOutput.id = null;
+      realOutputStats.id = null;
+      realOutputReport.id = null;
+    }
+  }
+
+  function stubOutput () {
+    removeRealOutput();
+    var tmp = document.createElement('div');
+    tmp.id = 'mocha';
+    document.body.appendChild(tmp);
+    return tmp;
+  }
+
+  function restoreOutput () {
+    var tmp = document.getElementById('mocha');
+    if (tmp && tmp !== realOutput) { tmp.parentNode.removeChild(tmp); }
+    if (realOutput) {
+      realOutput.id = 'mocha';
+      realOutputStats.id = 'mocha-stats';
+      realOutputReport.id = 'mocha-report';
+    }
+    return tmp !== realOutput ? tmp : undefined;
+  }
+
+  before(saveRealOutput); // so that the following afterEach is guaranteed to work
+
+  afterEach(function () { restoreOutput(); }); // WARNING: this is solely a fallback so *further* tests can be guaranteed to report correctly; these tests will not be reported correctly unless `restoreOutput` is called at the end of the test, before throwing any errors
+
+  it('should show an error if the tag is missing', function () {
+    try {
+      // TODO: Figure out why it's ok to add an element to display this error but not ok to just add the missing element in the first place.
```

Comment:
We can feel free to remove this comment and replace it with a new GitHub issue. I just needed to jot it down while it was on my mind but didn't want to get sidetracked from the work at hand.